### PR TITLE
LOA-382  Fix-Quest-accept

### PR DESCRIPTION
**Pull request by SebastianMaier03**

### CODE REVIEW

**Review by JohannesHausch — DISMISSED** [review]
See comment above.

File: app/src/main/java/de/hdmstuttgart/thelaendofadventure/ui/helper/MapHelper.kt [hunk 1]
```diff
@@ -56,12 +56,16 @@ class MapHelper(
...
-                    viewList.forEach { viewAnnotation ->
+                try {
+                    for (i in pointAnnotationList.indices) {
+                        val pointAnnotation = pointAnnotationList[i]
+                        val viewAnnotation = viewList[i]
                         if (pointAnnotation == clickedAnnotation) {
                             viewAnnotation.toggleViewVisibility()
                         }
                     }
+                } catch (@Suppress("TooGenericExceptionCaught")e: IndexOutOfBoundsException) {
```
[the remark this answers is not in the document]
Comment (SebastianMaier03) [reply app/src/main/java/de/hdmstuttgart/thelaendofadventure/ui/helper/MapHelper.kt:67]: no

**Comment by JohannesHausch** [comment]
@SebastianMaier03 can you explain why you think this is irrelevant.

**Review by JohannesHausch — DISMISSED** [review]
See comment above.

**Comment by SebastianMaier03** [comment]
Because currently it is needed so that the application does not crash when the issue that fixes the core problem works, it will be removed.emoved Because currently it is needed so that the application does not crash when the issue that fixes the core problem works, it will be removed.

**Comment by JohannesHausch** [comment]
> Because currently it is needed so that the application does not crash when the issue that fixes the core problem works, it will be removed.emoved Because currently it is needed so that the application does not crash when the issue that fixes the core problem works, it will be removed.

Couldn't we instead wait for the core issue to be resolved before accepting code with potentially unnecessary code smells?

Could we postpone implementation of the @Suppress annotation code section until the core issue is resolved to ensure that we develop a clean and robust solution without accepting unnecessary code smells?

We should consider the possibility of revising the code accordingly once the core issue is resolved, rather than temporarily using generic exception handling. This will help us ensure that the code is of high quality and facilitate future development efforts.

**Comment by SebastianMaier03** [comment]
Its not known when its fixed so to not stop development process would counterproductive

**Review by JohannesHausch — APPROVED** [review]
Mhmmm

**Pull request merged**